route/src/route_generator_worker.cpp

```diff
@@ -438,5 +442,77 @@ namespace route {
         return true;
     }
 
+bool RouteGeneratorWorker::crossTrackErrorCheck(const geometry_msgs::PoseStampedConstPtr& msg, lanelet::ConstLanelet current)
```

Comment:
Still looking at this, but I'm a bit skeptical of the logic here.

Reply:
I believe the original approach we discussed was to have the alert trip if the vehicle was off the route. The route object contains all the lane lets on a given road that are in the same direction as the shortest path as described here 
https://github.com/usdot-fhwa-stol/carma-platform/tree/develop/carma_wm#route-definition

So I would think we would simply need to check if the vehicle was inside any of the lanelets on the route (not just shortest path) and  if not check what the cross-track distance to the nearest lanelet was to see if action needed to be taken. This logic seems to compare against the shortest path rather than the whole route which means it would trigger when passing another vehicle.